public/js/controllers/AppController.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+var ResultsView = require('../views/ResultsView');
+var Houses = require('../collections/Houses');
+var vent = require('../events/Vent');
+
+var houses = new Houses();
+var resultsView = new ResultsView({ el: '#results', collection: houses });
+var jqxhr = houses.fetch();
+
+//Events to listen to:
+vent.on('allHouses:selected', showAllHouses);
+vent.on('addHome:selected', showAddForm);
+vent.on('house:selected', showHouse);
+vent.on('home:selected', showHome);
+vent.on('deleteHouse:selected', deleteHouse);
+
+//callback functions depending on event:
+function showHome() {
+  jqxhr.done(function () {
+    resultsView.showHome();
+  });
+}
+
+function showAllHouses() {
+  jqxhr.done(function () {
+    resultsView.showAllHouses();
+  });
+}
+
+//change to id as it's the id that is handed in in router
+function showHouse(options) {
+  var model = options.model;
+  jqxhr.done(function () {
+    //console.log("id ", model.id);
+    model = houses.findWhere({ _id: model.id });
```

Comment:
When this is called from the showHouse method in your appRouter (which happens if you load the `houses/:id` route from the server), you're passing in an id on the options object. That means model will be undefined, you'll get a script error, and the rest of the application will stop working. To fix this, I recommend referencing the id from the options object, then passing the id in your `handleClick` method inside the HouseItemView.